android/src/main/java/com/swmansion/gesturehandler/core/GestureUtils.kt

```diff
@@ -47,4 +55,101 @@ object GestureUtils {
   }
   fun coneToDeviation(angle: Double): Double =
     cos(Math.toRadians(angle / 2.0))
+
+  fun updateStylusData(stylusData: StylusData, event: MotionEvent) {
+    // On web and iOS 0 degrees means that stylus is parallel to the surface. On android this value will be PI / 2.
+    val altitudeAngle = (PI / 2) - event.getAxisValue(MotionEvent.AXIS_TILT).toDouble()
+    val pressure = event.getPressure(0).toDouble()
+
+    var orientation = event.getOrientation(0).toDouble()
+
+    // To get azimuth angle, we need to use orientation property. Orientation value range is [-PI, PI] (https://developer.android.com/develop/ui/compose/touch-input/stylus-input/advanced-stylus-features#orientation)
+    // To shift range into [0, 2PI], we add 2PI if orientation is less than 0.
+    if (orientation < 0) {
+      orientation += 2 * PI
+    }
+
+    // To get the same value that we get on web, we have to perform shift by PI/2.
+    // However, if orientation is greater than 3PI/2 we would get angles greater than 2PI. Therefore we simply subtract 3PI/2 in that case.
+    val azimuthAngle = if (orientation >= 3 * PI / 2) {
+      orientation - 3 * PI / 2
+    } else {
+      orientation + PI / 2
+    }
```

Comment:
Shouldn't it simply be `orientation - 3PI/2 (mod 2PI)`?

Reply:
Almost, as discussed it should be

$$(orientation + \frac{\pi}{2})\mod{2\pi}$$

Changed in [560c28a](https://github.com/software-mansion/react-native-gesture-handler/pull/3111/commits/560c28a2b0c07a04d6c075586f46e6f6c94a4f98).

Reply:
I'm pretty sure it's exactly the same since we are operating `(mod 2PI)` 😄

Reply:
You're right 😅